### PR TITLE
Adding loss and test cases.

### PR DESCRIPTION
**Pull request by rupalimits**
I have added the following loss -

1. Dice Loss
2. Jaccard Loss
3. Cross Entropy Loss
4. Hausdorff DT Loss

and their respective test cases.

To run the test case, please use the following command - 
include("test/TestLosses.jl")
on julia REPL

Screenshot 2025-04-09 at 1 59 19 PM

### CODE REVIEW

**Comment by jakubMitura14** [comment]
Ok generally looks good but couple things
1) use pycall to test against monai python library on random tensors not only edge cases and check weather results are approximately equal
2) Have you checked your implementations on cuda/amd to make sure that kernel wors as should
3) we are working on 2 and 3 dimensional data that has channel and is batched using the same convention as lux.jl so batch dimension last and channel one before; hence for 3d images we should work on 5 dimensional tensors and on 2d on 4 dimensional ones
Still considering how short you are working on it extremely good job !

**Comment by rupalimits** [comment]
In the recent commit, I have updated the implementation of the following loss function to match with the monai implementation - dice loss, Jaccard Loss and Cross Entropy Loss. The mentioned loss functions are also able to handle 4d and 5d tensors for 2d and 3d data. I have also added test cases to compare the output of current implementation with monai. 
To run the loss functions in your local, follow the below steps - 
1. Create a virtual env in the folder where you have cloned this repo with the help of following command - 
python3 -m venv .venv
source .venv/bin/activate
2. Install monai, numpy and torch in this venv to be able to test the losses against monai - 
pip install monai torch numpy scipy
3. Then on julid repl give the following command - 
include("test/TestLosses.jl")

The output will show you how the current implementation of loss functions is performing against monai.

**Comment by jakubMitura14** [comment]
Hello nice work but why are you sompering cross entropy loss in julia with dice loss in monai - and by the way this test passed? It should not as those are diffrent algorithms.
Also we need linearized way to get a batched algorithms working not looping over batch dimension as it will not use parallelism and adding batch is all about parallelism :)

**Comment by rupalimits** [comment]
I have updated the code according to your instructions. Got to learn about GPU programming in Julia - Thanks for that. I am sharing the output from colab where I have printed individual loss (though the print statement is not present in the commit).
Looking forward to your feedback.

Screenshot 2025-04-27 at 8 27 59 PM
